packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -525,6 +525,34 @@ export class PHPRequestHandler {
 			this.#cookieStore.rememberCookiesFromResponseHeaders(
 				response.headers
 			);
+
+			/**
+			 * Safari has a bug that prevents Service Workers from redirecting relative URLs.
+			 * When attempting to redirect to a relative URL, the network request will return an error.
+			 * See the Webkit bug for details: https://bugs.webkit.org/show_bug.cgi?id=282427.
+			 *
+			 * Because PHP and WordPress can redirect to both relative and absolute URLs
+			 * using the `location` header we need to ensure redirects are processed
+			 * correctly by the Service Worker.
+			 *
+			 * As a workaround for Safari Service Workers, we convert the `location` header
+			 * to an absolute URL for all redirect responses (300-399 status codes).
+			 * This resolves the issue because Safari correctly handles redirects
+			 * to absolute URLs provided by the `location` header.
+			 */
+			if (
+				response.httpStatusCode.toString().startsWith('3') &&
+				response.headers['location'] &&
+				!response.headers['location'][0].startsWith(this.#PROTOCOL)
+			) {
+				response.headers['location'] = [
+					[
+						`${this.#PROTOCOL}://`,
+						this.#HOST,
+						response.headers['location'],
```

Comment:
It looks like we still need to account for redirects that are relative to the request URI.

For example, a request to `https://example.com/examples-club/forum` could respond with
`Location: login`
which should ultimately resolve to
`https://example.com/examples-club/login`

Currently, it looks like this code would resolve that to
`https://example.comlogin`

For reference from https://www.rfc-editor.org/rfc/rfc7231#section-7.1.2

> The "Location" header field is used in some responses to refer to a
> specific resource in relation to the response.  The type of
> relationship is defined by the combination of request method and
> status code semantics.
> 
>   Location = URI-reference
> 
> The field value consists of a single URI-reference.  When it has the
> form of a relative reference ([[RFC3986], Section 4.2](https://www.rfc-editor.org/rfc/rfc3986#section-4.2)), the final value
> is computed by resolving it against the effective request URI
> ([[RFC3986], Section 5](https://www.rfc-editor.org/rfc/rfc3986#section-5)).